pkg/backends/calico/client.go

```diff
@@ -421,14 +441,21 @@ func (c *client) updatePeersV1() {
 					peers = append(peers, c.nodeAsBGPPeers(peerNodeName)...)
 				}
 			} else {
-				ip := cnet.ParseIP(v3res.Spec.PeerIP)
+				// Separate port from Ip if it uses <ip>:<port> format
+				host, port := parseIPPort(v3res.Spec.PeerIP)
+				ip := cnet.ParseIP(host)
 				if ip == nil {
 					log.Warning("PeerIP is not assigned or is malformed")
 					continue
 				}
+				if port == "" {
+					// If this peer IP belongs to a calico/node that has listen port set, use that port to peer
+					port = c.getListenPortForIP(ip.String())
+				}
 				peers = append(peers, &bgpPeer{
 					PeerIP: *ip,
 					ASNum:  v3res.Spec.ASNumber,
+					Port:   port,
```

Comment:
If we are extending GlobalBGPPeerKey & NodeBGPPeerKey to include port, so I assume keys will look something like `/calico/bgp/v1/global/peer_v4/x.x.x.x/<port>` and `/calico/bgp/v1/host/<node_name>/peer_v4/x.x.x.x/<port>` and most of the changes will involve updating the bird templates to handle the new key. Is that correct?

I can work on it in a separate PR, if that is the way we are going.

Reply:
>  GlobalBGPPeerKey and NodeBGPPeerKey so as to disambiguate between peers with the same IP but different ports?

Hm, I suppose the data model has already supported multiple BGP peers with the same IP address (though that is a very odd scenario, so wouldn't expect real use-cases for that). 

I'd guess if a user did _that_ today, it would result in inconsistent behavior since we'd resolve two v3 objects to the same v1 object? 

Seems reasonable to add the port to the v1 keys now, though.

Reply:
Thanks @Suraiya-Hameed and @caseydavenport .  Doesn't sound like we're completely sure about this yet; but I'm happy with deferring this to another story/PR.  Suraiya, please can you raise a story to track?

On the detail, I would recommend `.../peer_v4/x.x.x.x-<port>` rather than `.../peer_v4/x.x.x.x/<port>`, so the key is still at the same place in the etcd tree, as that's more likely to work with the existing templates.

Reply:
Thanks! I have created a story to track that.